core/solver/bicgstab.cpp

```diff
@@ -111,6 +111,8 @@ void Bicgstab<ValueType>::apply_dense_impl(
 
     auto exec = this->get_executor();
 
+    Array<char> reduction_tmp{exec};
```

Comment:
move it to mutable class member? in case we need reusing these in the multigrid or something else.

Reply:
That should be handled later IMO, together with #805.